### ISSUE TITLE
关于扫描数据的测试

### ISSUE THREAD

**Issue by qiji77** [wangsen1312/unsupervised3dhuman#1]
你好，请原谅我英文不太好，怕表达的不清晰，就用中文说了。
我测试了您的网络。用的是FAUST的扫描模型。
请问是需要对这里的扫描模型进行预处理么？我直接用的话，得到的结果好像有些差，还是说我的环境有什么问题，我直接用的readme上的提示做的，创建的虚拟环境。
这是我的结果
![image](https://user-images.githubusercontent.com/30361443/132829514-70406e3a-1473-4e1f-87f8-f1618ee18213.png)
![image](https://user-images.githubusercontent.com/30361443/132829551-2bb2cfbc-e106-4d1e-b345-ead0dd4aa405.png)
![image](https://user-images.githubusercontent.com/30361443/132829672-f3fbb819-6263-41e0-8d70-4ce9cdb017d8.png)
![image](https://user-images.githubusercontent.com/30361443/132829711-de3b3fe9-0901-407d-8237-60f1390b3d26.png)
分别对应的scans中的096和035.
希望能得到您的帮助，不胜感激

**Comment by wangsen1312** [comment]
@qiji77  那我也就回复中文给你
首先感谢你的测试
我们提供的demo code 主要是针对CAPE数据集的， 其中训练也仅仅只是在CAPE上面做的。
针对FAUST这个数据集 我们是做了一些后处理的~
1. 针对性别 我们选定了 男女 并非使用的中性模型
2. 因为faust的一些姿势并非属于常见姿势，我们对权重进行了一些改变保证最终的形变可以达到该姿势~ 
3. 我们还使用了non-registration做了一些处理

仅仅使用我给的demo code，个人觉得很多case应该是可以的， 部分case 做不好也属于正常的~

如果还有什么问题可以继续交流

**Comment by qiji77** [comment]
好的，感谢您的回复，我明白了。
不知道您是否方便提供一下FAUST的修补后模型，我也想用···
之前直接做的效果并不是很好，我的邮箱是775786698@qq.com
如果不方便就算了，总之 还是谢谢您

**Comment by wangsen1312** [comment]
@qiji77 我给你回复邮件了，咱们可以邮件单独聊。

**Comment by qiji77** [comment]
阿西，是775785698@qq.com  shit  我说怎么没看到邮件  抱歉抱歉

**Comment by YJonmo** [comment]
> 

We have added a loss term that prevents the leg crossing:
leg_preserve_loss = (10 ** 6) * ((body_pose[:, 2] - 0.0) ** 2).sum(dim=-1) + \
                        (10 ** 6) * ((body_pose[:, 5] + 0.0) ** 2).sum(dim=-1)

total_loss = correspond_loss + pose_prior_loss + shape_prior_loss + betas_preserve_loss + \
                 pose_preserve_loss + chamfer_loss + point2mesh_loss + leg_preserve_loss

However, this loss term assumes the person's legs are not changing much.

**Comment by wangsen1312** [comment]
@YJonmo Thank you for your information. Actually we have considered to add some strong prior to prevent extreme pose. 
The main problem is I have met some test cases in CAPE/CMU dataset has extreme pose, so we didn't add too much prior here. 
I think it is may be a good idea to calculate the final loss with/without such strong priors, and use the smaller as final one.
